packets/packets.go

```diff
@@ -401,28 +463,36 @@ func ReadPacket(data io.Reader) (p *Packet, err error) {
 			switch p.format.dtype[0] {
 			case reflect.Int16:
 				result := make([]int16, p.payloadLength/2)
-				if err = binary.Read(data, p.format.endian, result); err != nil {
+				bslice := getbytes.FromSliceInt16(result)
+				if _, err = io.ReadFull(data, bslice); err != nil {
```

Comment:
I think this is identical to `bytesToRawType` except special cased for `Int16`. If so, I think it is worse in the following ways
1. more verbose, it would take 2 line to do what is 5 lines here
2. I don't believe an error is possible, the code currently suggests it is
3. the current code is probably slower

Reply:
`bytesToRawType` does not read from an `io.Reader` like this does. So the number of lines compared is not fair, and that's also where the error is possible. Also, packets is a stand-alone sub-package that doesn't know anything about `RawType`. Finally, if `RawType` were ever compiled to be 32 bits (oh, the nightmare), then it would no longer match this case.

Reply:
I see, I misunderstood this, I thought you were creating the reader just to call ReadFull on it, but I think this makes sense.